src/cargo-kani/src/main.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::Result;
+use call_cargo::config_toml_to_args;
+use std::ffi::OsString;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use util::alter_extension;
+
+mod args;
+mod call_cargo;
+mod call_cbmc;
+mod call_cbmc_formatter;
+mod call_cbmc_viewer;
+mod call_goto_cc;
+mod call_goto_instrument;
+mod call_single_file;
+mod call_symtab;
+mod session;
+mod util;
+
+fn main() -> Result<()> {
+    match determine_invocation_type(Vec::from_iter(std::env::args_os())) {
+        InvocationType::CargoKani(args) => cargokani_main(args),
+        InvocationType::Standalone => standalone_main(),
+    }
+}
+
+fn cargokani_main(mut input_args: Vec<OsString>) -> Result<()> {
+    input_args.extend(config_toml_to_args()?);
+    let args = args::CargoKaniArgs::from_iter(input_args);
+    args.validate();
+    let ctx = session::KaniSession::new(args.common_opts)?;
+
+    let symtabs = ctx.cargo_build()?;
+    if ctx.args.only_codegen {
+        return Ok(());
+    }
+    let mut goto_objs: Vec<PathBuf> = Vec::new();
+    for symtab in &symtabs {
+        goto_objs.push(ctx.symbol_table_to_gotoc(symtab)?);
```

Comment:
This is a candidate for parallelization (in a future PR).

Reply:
Yep! There are a couple others too. For instance, the multiple runs of cbmc when invoking viewer.